src/melee.cpp

```diff
@@ -1353,6 +1353,33 @@ bool player::can_weapon_block() const
             weapon.has_technique( WBLOCK_3 ));
 }
 
+int blocking_ability( const item &shield )
+{
+    int block_bonus = 2;
+    if (shield.has_technique( WBLOCK_3 )) {
+        block_bonus = 10;
+    } else if (shield.has_technique( WBLOCK_2 )) {
+        block_bonus = 6;
+    } else if (shield.has_technique( WBLOCK_1 )) {
+        block_bonus = 4;
+    }
+    return block_bonus;
+}
+
+item &player::best_shield()
+{
+    int weapon_block = blocking_ability( weapon );
+    for( item &shield : worn ) {
+        if( shield.has_flag( "BLOCK_WHILE_WORN" ) && blocking_ability( shield ) >= weapon_block ) {
+            return shield;
+        }
+    }
+    if( can_weapon_block() ) {
+        return weapon;
+    }
+return ret_null;
```

Comment:
Format this function.
We don't have melee.cpp astyled yet (or it would be a failed build), but this isn't an excuse to leave the code unformatted.